### PR TITLE
Syncval: Leak fix of circular share_ptr between swapchains and queue batch contexts

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Fixes: #5382

### CODE REVIEW

File: tests/framework/layer_validation_tests.cpp [hunk 1]
```diff
@@ -1110,13 +1110,7 @@ void VkLayerTest::AddSurfaceExtension([[maybe_unused]] const WsiPreference prefe
 
     // User doesn't care which WSI backend they use. Pick the first one that works
     assert(preference == WsiPreference::Default);
-#if defined(VK_USE_PLATFORM_WAYLAND_KHR)
-    if (auto temp_display = wl_display_connect(nullptr); temp_display) {
-        AddRequiredExtensions(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME);
-        wl_display_disconnect(temp_display);
-        return;
-    }
-#endif
```
Comment (juan-lunarg) [inline tests/framework/layer_validation_tests.cpp:1119]: See https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/5482

**Review by charles-lunarg — APPROVED** [review]
Looks decent to me, though I mainly focused on the leak aspect, which seems to be working as expected. I don't have much experience with Sync so I'll others with more knowledge speak on that aspect.

**Review by juan-lunarg — APPROVED** [review]
LGTM

**Review by ncesario-lunarg — APPROVED** [review]

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

**Pull request merged**